web/src/app/util/avatars.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react'
+import { Layers, RotateRight, Today, VpnKey, Person } from '@material-ui/icons'
+import { useSessionInfo } from './RequireConfig'
+import { Avatar, AvatarTypeMap, SvgIconProps } from '@material-ui/core'
+import { OverridableComponent } from '@material-ui/core/OverridableComponent'
+
+type IconProps = (props: SvgIconProps) => JSX.Element //type alias
+interface AvatarProps extends OverridableComponent<AvatarTypeMap> {
+  userID?: string
```

Comment:
This is only used for the `UserAvatar` and is required. We should define it separately from the others.